core/domain/collection_domain.py

```diff
@@ -186,27 +225,52 @@ def to_dict(self):
 
     @classmethod
     def from_dict(cls, node_dict):
+        """Return a CollectionNode domain object from a dict.
+
+        Args:
+            node_dict: dict. The dict representation of CollectionNode object.
+
+        Returns:
+            A CollectionNode domain object.
+        """
         return cls(
             copy.deepcopy(node_dict['exploration_id']),
             copy.deepcopy(node_dict['prerequisite_skills']),
             copy.deepcopy(node_dict['acquired_skills']))
 
     @property
     def skills(self):
-        """Returns a set of skills where each prerequisite and acquired skill
-        in this collection node is represented at most once.
+        """Returns a set of skills.
+
+        Returns:
+            The union of prerequisete set and acquired skill set. The skills
```

Comment:
Missing type info?

set(str). The union of the prerequisite and acquired skills. Each skill is represented at most once.